src/shared/util.js

```diff
@@ -454,6 +454,8 @@ function bytesToString(bytes) {
 }
 
 function stringToBytes(str) {
+  assert(str !== null && str !== undefined,
```

Comment:
Make it: assert(typeof str !== 'string', ...);

Also bytesToString looks non-symmetrical now :) can we add asset(bytes === null || typeof bytes !== 'object' || bytes.length === undefined, ...);